uxarray/grid/grid.py

```diff
@@ -231,6 +231,24 @@ def from_dataset(
 
         return cls(grid_ds, source_grid_spec, source_dims_dict)
 
+    @classmethod
+    def from_geodataframe(
+        cls,
+        filename: str,
+        **kwargs,
+    ):
+        """Constructs a ``Grid`` object from a shapefile.
+
+        Parameters
+        ----------
+        filename : str
+           shapefile filename
```

Comment:
Mention that any data that is supported by `geopandas.read_file()` is supported here.

https://geopandas.org/en/stable/docs/reference/api/geopandas.read_file.html#geopandas-read-file